Re-Enable CA1416

PureWeen opened the pull request:
### Description of Change

<!-- Enter description of the fix in this section -->

### Issues Fixed

<!-- Please make sure that there is a bug logged for the issue being fixed. The bug should describe the problem and how to reproduce it. -->

Fixes #

<!--
Are you targeting main? All PRs should target the main branch unless otherwise noted.
-->

PureWeen committed:
Re-Enable CA1416

jsuarezruiz commented on src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs:
```diff
@@ -27,17 +28,36 @@ public static UIImage ScaleImage(this UIImage target, float maxWidth, float maxH
-			UIGraphics.EndImageContext();
-
-			if (disposeOriginal)
+			if (!OperatingSystem.IsIOSVersionAtLeast(17))
```
Require this validation on more places (essentials etc):

`D:\a\_work\1\s\src\Essentials\src\Screenshot\Screenshot.ios.cs(89,4): error CA1416: This call site is reachable on: 'MacCatalyst' 13.1 and later. 'UIGraphics.EndImageContext()' is unsupported on: 'maccatalyst' 17.0 and later. `

stephen-hawley commented on src/Core/src/Handlers/SwipeItemMenuItem/SwipeItemMenuItemHandler.iOS.cs:
```diff
@@ -142,12 +142,32 @@ static UIImage MaxResizeSwipeItemIconImage(UIImage sourceImage, nfloat maxWidth,
+					Scale = 0
+				};
+
+				using (var renderer = new UIGraphicsImageRenderer(new CGSize(width, height), format))
```
minor - there's no reason to use a block which you can just do:
```
using var renderer = new UIGraphicsImageRenderer(new CGSize(width, height), format);
```

stephen-hawley commented on src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs:
```diff
@@ -27,17 +28,36 @@ public static UIImage ScaleImage(this UIImage target, float maxWidth, float maxH
-			return image;
+			using (var renderer = new UIGraphicsImageRenderer(target.Size))
+			{
+				var resultImage = renderer.CreateImage((UIGraphicsImageRendererContext imageContext) =>
```
same

stephen-hawley commented on src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs:
```diff
@@ -47,17 +67,35 @@ public static UIImage NormalizeOrientation(this UIImage target, bool disposeOrig
 			}
 
-			return image;
+			using (var renderer = new UIGraphicsImageRenderer(target.Size))
```
same

stephen-hawley committed:
remove CA1416 errors

stephen-hawley committed:
whitespace issues.

PureWeen commented:
/rebase

PureWeen commented:
/rebase

PureWeen commented:
/rebase

stephen-hawley committed:
Fixed all remaining CA1416 errors

PureWeen committed:
Fixed all remaining CA1416 errors (#22868)

This should take care of all the remaining CA1416 errors.

Here's what I found about the analyzer for this -
* It understands when code from Xamarin/macios is reachable from maui
code and will flag an error
* It never understands the pattern `if
(!OperatingSystem.IsZZZZZZVersionAtLeast(major, minor)) return;` that
implies that the rest of the code in the method is safe, flagging a
false failure
* Compound OS tests or complex code inside the test will flag a false
failure
* In the case of [this
code](https://gist.github.com/stephen-hawley/716761d318fbcddd7795ed53d331cdc8),
the analyzer appears to hang, although to be clear, I don't have
evidence that it is the analyzer doing this

In order to work around this, the simplest way is to refactor all the OS
version specific code into its own method and flag it with
`[SupportedOSPlatform("xxxyy.zz")]`. In this case, the above false
failures go away.

In rare cases, I was unable to do this and instead used a `#pragma` to
suppress the warning around the code that triggered the false failure. I
tried to avoid this at all costs since this is source of bugs waiting to
happen.

stephen-hawley committed:
Merge branch 'main' into reenable_CA1416

stephen-hawley committed:
warnings ignored on compatability (#23189)

PureWeen committed:
Merge branch 'main' into reenable_CA1416

PureWeen committed:
Update Compatibility.csproj

PureWeen commented:
/azp run

PureWeen committed:
Merge remote-tracking branch 'origin/main' into reenable_CA1416

PureWeen committed:
- fix CA1413
